docs/kuneiform/procedures.mdx

```diff
@@ -0,0 +1,336 @@
+---
+sidebar_position: 3
+sidebar_label: "Procedures"
+id: kuneiform-procedures
+title: Kuneiform Procedures
+description: An outline of Kuneiform procedures.
+slug: /kuneiform/procedures
+---
+
+Kuneiform supports a basic procedural language for building logic into schemas. Procedures can be used for building access
+control logic, performing arithmetic, and managing control flow.
+
+## Declaring Procedures
+
+Procedures are declared using the `procedure` keyword. They can be given a list of parameters and their corresponding types,
+return types, and access modifiers.
+
+The example below shows the signature for a `get_user` procedure. It takes one input (the id of the user, of type `uuid`),
+and returns the user's name and age. It will return exactly one record, containing a name and age:
+
+```ts
+// get_user takes one UUID, and will return a single name and age.
+// It has a view tag, which means it does not mutate state, and can
+// be called without submitting a blockchain transaction
+procedure get_user($id uuid) public view returns (name text, age int) {
+    // procedure implementation
+}
+```
+
+Procedures can also return tables, which are a collection of many records. The below example returns all user IDs and names
+of a certain age:
+
+```ts
+// get_users takes one integerm and will return a table with two columns:
+// "name" of type text, and "age" of type int
+// It has a view tag, which means it does not mutate state, and can
+// be called without submitting a blockchain transaction
+procedure get_users($age int) public view returns table(name text, age int) {
+    // procedure implementation
+}
+```
+
+If a procedure does not return anything, the return should not be specified:
+
+```ts
+// create_user takes a text and int, and returns nothing.
+// It does not have a view tag so that it can mutate state,
+// and therefore can only be called with a blockchain transaction.
+procedure create_user($name text, $age int) public {
+    // procedure implementation
+}
+```
+
+## Access Modifiers
+
+Procedures can be given access modifiers to specify how/when they can be called. All procedures must be specified
+as `public` or `private`: `public` procedures can be called externally, while `private` procedures can only be called
```

Comment:
I think it is easier to read if the access modifiers are presented in a table. The way this reads, it is hard to follow all the supported tags.

Reply:
I sort've disagree, because a table conveys that they are all options. For example, in the table, since `view` and `owner` are there, it makes me think that I can use those instead of public. We even saw fractal trip up on this. I think this more clearly conveys that `owner` and `view` are not replacements for public and private